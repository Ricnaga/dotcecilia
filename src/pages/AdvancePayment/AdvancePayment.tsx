import {
  PageContainer,
  PaymentContainer,
  PaymentTable,
} from '@shared/components';
import { useInputFields } from '@shared/hooks/useInputFields';
import { useReactPrint } from '@shared/hooks/useReactPrint';
import { useMemo } from 'react';
import {
  AdvancePaymentTableBody,
  AdvancePaymentCalculator,
  AdvancePaymentCalculatorFields,
} from './components';

export function AdvancePayment() {
  const {
    data: { printRef },
    functions: { setPrint },
  } = useReactPrint();

  const {
    data: { values },
    functions: { onChangeValue },
  } = useInputFields<AdvancePaymentCalculatorFields>({
    name: '',
    salary: 0,
    refDate: new Date(),
  });

  const bodyMemoized = useMemo(
    () => <AdvancePaymentTableBody salary={values.salary} />,
    [values.salary],
  );

  return (
    <PageContainer title="Vale adiantamento">
      <PaymentContainer
        calculator={
          <AdvancePaymentCalculator
            onChange={onChangeValue}
            onPrint={setPrint}
            values={values}
          />
        }
        printRef={printRef}
      >
        <PaymentTable
          headerType="ADIANTAMENTO"
          name={values.name}
          monthRef={values.refDate}
        >
          {bodyMemoized}
        </PaymentTable>
      </PaymentContainer>
    </PageContainer>
  );
}
