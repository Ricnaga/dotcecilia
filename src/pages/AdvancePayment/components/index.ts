export * from './AdvancePaymentTableBody/AdvancePaymentTableBody';
export * from './AdvancePaymentCalculator/AdvancePaymentCalculator';
