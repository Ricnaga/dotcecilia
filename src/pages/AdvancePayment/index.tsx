import { CeciliaPageHeader, CeciliaTable } from '../../shared/components';
import { AdvancePaymentTableBody } from './components';

export function AdvancePaymentPage() {
  return (
    <CeciliaPageHeader title="Vale adiantamento">
      <div className="grid grid-cols-12 gap-10">
        <div className="grid grid-cols-12 col-span-7">
          <CeciliaTable headerType="ADIANTAMENTO">
            <AdvancePaymentTableBody />
          </CeciliaTable>
        </div>
        <div className="col-span-5">
          <h1>oi</h1>
        </div>
      </div>
    </CeciliaPageHeader>
  );
}
