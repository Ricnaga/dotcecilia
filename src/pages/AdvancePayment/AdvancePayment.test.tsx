import { render } from '@testing-library/react';
import { AdvancePaymentPage } from './AdvancePayment';

describe('Page: AdvancePayment', () => {
  it('should render correctly', () => {
    const { container } = render(<AdvancePaymentPage />);
    expect(container).toMatchSnapshot();
  });
});
