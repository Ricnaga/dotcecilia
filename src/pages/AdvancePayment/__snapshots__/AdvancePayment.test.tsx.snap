// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: AdvancePayment should render correctly 1`] = `
<div>
  <div
    class="w-full"
  >
    <p
      class="text-5xl font-bold my-6"
    >
      Vale adiantamento
    </p>
    <div
      class="grid grid-cols-12 gap-10"
    >
      <div
        class="grid grid-cols-12 col-span-7"
      >
        <table
          class="col-span-12 h-1 bg-slate-100 border border-slate-800 text-slate-900"
        >
          <thead
            class=""
          >
            <tr>
              <th
                colspan="6"
              >
                undefined
              </th>
              <th
                colspan="2"
              >
                RECIBO DE 
                ADIANTAMENTO
              </th>
            </tr>
          </thead>
          <thead>
            <tr>
              <th
                class="text-right"
                colspan="6"
              >
                undefined
              </th>
              <th
                colspan="2"
              >
                Ref. SETEMBRO
              </th>
            </tr>
            <tr>
              <th
                colspan="8"
              >
                CNPJ: 
                undefined
              </th>
            </tr>
            <tr>
              <th
                colspan="8"
              >
                <p
                  class="my-4 border-y border-slate-800"
                >
                  Nome
                </p>
              </th>
            </tr>
          </thead>
          <tbody
            class="border-t border-slate-800"
          >
            <tr>
              <td
                class="border-r border-slate-800 undefined"
              >
                SALÁRIO BASE
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="3"
              >
                 0,00
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                Ref.
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                vencimento
              </td>
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                class="border-r border-slate-800 undefined"
                colspan="5"
              >
                <p
                  class="mb-20"
                >
                  Adiantamento:
                </p>
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                <p
                  class="mb-20"
                >
                  40
                </p>
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                <p
                  class="mb-20 text-center"
                >
                   
                   0,00
                </p>
              </td>
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                colspan="5"
              >
                <p
                  class="my-4"
                >
                  Total líquido:
                </p>
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="3"
              >
                 0,00
              </td>
            </tr>
          </tbody>
          <tfoot
            class="border-t border-slate-800"
          >
            <tr>
              <td
                colspan="4"
              >
                <p
                  class="my-8 text-center"
                >
                  Assinatura:.......................................................
                </p>
              </td>
              <td
                colspan="4"
              >
                <p
                  class="my-8 text-center"
                >
                  Data:__/__/__
                </p>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
      <div
        class="col-span-5"
      >
        <div
          class="grid grid-cols-12"
        >
          <p
            class="text-4xl text-left font-bold mb-4 col-span-6"
          >
            Cálculos
          </p>
          <button
            class="btn btn-yellow col-start-9 col-end-13 mb-4"
            type="button"
          >
            Imprimir
          </button>
        </div>
        <div
          class="grid grid-cols-12 gap-4"
        >
          <div
            class="col-span-12"
          >
            <label
              class="text-xl"
              for="text"
            >
              <input
                placeholder="Digite o nome"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="col-span-6"
          >
            <label
              class="text-2xl"
              for="text"
            >
              <input
                min="0.00"
                placeholder="salário"
                step="0.01"
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="col-span-6"
          >
            <label
              class="text-xl"
              for="month"
            >
              <input
                type="month"
                value="2022-10"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
